source/grid/grid_in.cc

```diff
@@ -2076,7 +2076,9 @@ void GridIn<3>::read_netcdf (const std::string &filename)
   // take numbers::internal_face_boundary_id
   // as it denotes an internal face
   for (unsigned int i=0; i<bmarker.size(); ++i)
-    Assert(0<=bmarker[i] && bmarker[i]<numbers::internal_face_boundary_id, ExcIO());
+    Assert(0<=bmarker[i] &&
+           static_cast<types::boundary_id>(bmarker[i])<numbers::internal_face_boundary_id,
```

Comment:
The question is whether bmarker or boundary_id is the bigger type. If bmarker is the bigger type, casting to the bigger type and "<" is correct. This was useful before, where invalid_boundary_id can be 255 while we could be reading in something bigger (which is a mistake).